pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -255,16 +261,49 @@ func Convert_v1_Disk_To_api_Disk(c *ConverterContext, diskDevice *v1.Disk, disk
 	return nil
 }
 
-func checkDirectIOFlag(path string) bool {
-	// check if fs where disk.img file is located or block device
-	// support direct i/o
-	// #nosec No risk for path injection. No information can be exposed to attacker
-	f, err := os.OpenFile(path, syscall.O_RDONLY|syscall.O_DIRECT, 0)
-	if err != nil && !os.IsNotExist(err) {
-		return false
+type DirectIOChecker interface {
```

Comment:
On the long run we should do the directio checking outside of the converter and pass the result via the convertercontext.  See for instance also #5944. So that the converter can fully focus on the conversion and is not concerned with discovery.

Reply:
So basically it means moving the directio checking from `converter.SetDriverCacheMode` and making smth like:

```go
	// set drivers cache mode
	for i := range domain.Spec.Devices.Disks {
                supportsDirectIO := checkDirectIOForDisk(&domain.Spec.Devices.Disks[i]) // <--- this part is taken out from SetDriverCacheMode
		err := converter.SetDriverCacheMode(&domain.Spec.Devices.Disks[i], supportsDirectIO)
		if err != nil {
			return domain, err
		}
		converter.SetOptimalIOMode(&domain.Spec.Devices.Disks[i])
	}
```
Correct?

Reply:
More like what we do with `IsBlockPVC` and `IsBlockDV`. We would fill the `ConverterContext` which would then contain the info when calling `Convert_v1_VirtualMachineInstance_To_api_Domain(vmi, domain, context)`.

Reply:
Okay. Initially directio checking was implemented within the conveter so I did not think about moving it from there. But I think it makes sense considering the split of responsibilities. I can probably prepare a follow-up PR to address that.